src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -535,6 +535,18 @@ public SymbolInfo GetSymbolInfo(ExpressionSyntax expression, CancellationToken c
 
                 return SymbolInfo.None;
             }
+            else if (SyntaxFacts.IsDeconstructionType(expression, out parent))
+            {
+                Debug.Assert(((VariableDeclarationSyntax)parent).Variables.Count == 1);
+                TypeSymbol deconstructionType = (GetDeclaredSymbol(((VariableDeclarationSyntax)parent).Variables.First(), cancellationToken) as LocalSymbol)?.Type;
+
+                if (deconstructionType?.IsErrorType() == false)
+                {
+                    return new SymbolInfo(deconstructionType);
+                }
+
+                return SymbolInfo.None;
```

Comment:
It looks like this code is pretty much a duplication of the previous block. Consider extracting into a method and passing target declarator as a parameter.